cisco_ise/actions/lib/base.py

```diff
@@ -0,0 +1,458 @@
+            sys.exit(1)
+
+        # Check if formatting details were passed
+        # if not, check if self.payload_format exists and use that value
+        # If it doesn't exist, just set it to JSON
+        if not payload_format:
+            try:
+                self.payload_format
+            except NameError:
+                payload_format = 'json'
+            else:
+                payload_format = self.payload_format
+
+        # Validate payload_format
+        if payload_format not in ['json', 'xml']:
+            self.logger.error(
+                f'Valid types for payload_format are ["json", "xml"]. Received "{payload_format}"')
+            sys.exit(1)
+
+        headers = {
+            'Accept': 'application/json',
+            'Content-Type': f'application/{payload_format}'
+        }
+
+        auth = HTTPBasicAuth(api_user, api_pass)
+
+        # GET and DELETE don't need a payload sent
+        # Sometimes a GET will be a 'Get-All' which is effectively a search.
+        # In that case, results need to be handled variably
+        if http_method in ['GET', 'DELETE']:
+            result = []
+            last = False
+
+            # Run the api_request() on a loop until we have no more pages
+            # page['SearchResult']['nextPage']
+            while True:
+                page, response_code, succeeded, response_msg = \
+                    self.api_request(http_method,
+                                     request_uri,
+                                     auth=auth,
+                                     headers=headers,
+                                     verify_ssl=verify_ssl)
+
+                # Check if there is a 'SearchResult'
+                # 'SearchResult' is present on a 'Get-All' kind of request
+                # and will contain a list of results and potentially a 'nextPage'
+                if 'SearchResult' in page:
+                    # Check if there is a 'nextPage'
+                    if 'nextPage' in page['SearchResult']:
+                        request_uri = page['SearchResult']['nextPage']['href']
+                    else:
+                        last = True
+
+                    # If there was 'SearchResult', 'resources' should have the data we care about
+                    if 'resources' in page['SearchResult']:
+                        result.extend(page['SearchResult']['resources'])
+
+                # Looks like this isn't a 'Get-All' so there is probably just 1 key in page
+                else:
+                    result = page
+                    last = True
+
+                # All pages have been processed, return the applicable result and break the loop
+                if last:
+                    return self.format_action_result(result,
+                                                     response_code,
+                                                     succeeded,
+                                                     response_msg)
+                    break
```

Comment:
```suggestion
                    return self.format_action_result(result,
                                                     response_code,
                                                     succeeded,
                                                     response_msg)
```
`break` is not needed after `return`.

Reply:
good catch, cruft left over from a loop rewrite.